### ISSUE TITLE
Opening video from nautilus sometimes opens player with empty playlist

### ISSUE THREAD

**Issue by rokm** [gnome-mpv/gnome-mpv#377]
I am hitting another issue, which again occurs only sporadically. When opening a video file via double-click in nautilus, gnome-mpv sometimes opens an empty player window instead, with empty playlist.

This seems to be connected to the state of mpv's properties, in particular ```idle-active```.

On successful runs, I can see ```idle-active``` first change to 1; then eventually ```playlist``` is changed, and we skip the [if block](https://github.com/gnome-mpv/gnome-mpv/blob/b67331ba74b68f6bd991b35f7066b3ed23caaa4c/src/gmpv_player.c#L321) due to idle-active being one. Then ```vo-configured``` is changed and playlist is loaded from ```player->playlist``` to GmpvMpv object via ```load_from_playlist```.

On an unsuccessful run, ```idle-active``` first changes to 0. So when ```playlist``` changes, we enter the [if block](https://github.com/gnome-mpv/gnome-mpv/blob/b67331ba74b68f6bd991b35f7066b3ed23caaa4c/src/gmpv_player.c#L321), which in turn attempts to read the playlist from GmpvMpv into ```player->playlist```. However, as the playlist in GmpvMpv object is empty at that point, we lose our playlist. So when ```vo-configured``` changes, we have empty playlist, and consequently nothing to play.

The work-aroud below seems to alleviate the issue (extending [the if block](https://github.com/gnome-mpv/gnome-mpv/blob/b67331ba74b68f6bd991b35f7066b3ed23caaa4c/src/gmpv_player.c#L321)):

```c
if(!idle_active)
{
    if (!player->init_vo_config) {
        update_playlist(player);
    } else {
        g_print("Enabling work-around!\n");
        was_empty = TRUE;
    }
}
```

With the above work-around, we attempt to pull the playlist from GmpvMpv object only after vo has been configured. This fixes the loss of the playlist.

Note that we check the player's variable that is toggled on ```vo-configured``` change instead of querying the property directly - my tests indicate that at that point, the property may be already set to 1, but we have not received the property change notification (ordering issue with notifications, perhaps?).

Resetting ```was_empty``` is necessary to start the playback.

I am not sure what the proper fix would be here. Nor if the issue is with gnome-mpv or with mpv itself (using 0.29.0 on Fedora 28). Perhaps someone with better knowledge can shed some light on this?

**Comment by gnome-mpv** [comment]
I think I have seen this happen before, but it happens so rarely on my machine that I never got around to fixing it.

Does the workaround prevent the issue from happening at all or just make it happen less often?

**Comment by rokm** [comment]
I haven't seen an empty window for the past two days during testing (while the added message about the enabled work-around did appear couple of times).

For this particular issue, I think the work-around should fix it completely. Because at its core, the issue is that on startup, the playlist is transferred from gnome-mpv to mpv only after vo is initialized, so before that happens, we need to avoid the transfer in opposite direction. 

What puzzles me is why does ```idle-active``` sometimes change to 0 during the startup. And similarly, why is ```playlist``` change signaled but the actual playlist on the mpv side (which, if I understand correctly, is where these properties are coming from) is empty...

**Comment by adrianheine** [comment]
I regulary have this issue with 0.16 when enqueuing files mounted through gvfsd-fuse.
